pkg/snap/util/calico.go

```diff
@@ -18,10 +18,23 @@ import (
 // Optionally, the new manifest may be applied using the microk8s-kubectl.wrapper script.
 func MaybePatchCalicoAutoDetectionMethod(ctx context.Context, s snap.Snap, canReachHost string, apply bool) error {
 	config, err := s.ReadCNIYaml()
+	var newConfig string
 	if err != nil {
 		return fmt.Errorf("failed to read existing cni configuration: %w", err)
 	}
-	newConfig := strings.ReplaceAll(config, `"first-found"`, fmt.Sprintf(`"can-reach=%s"`, canReachHost))
+	if strings.Count(canReachHost, ":") > 0 {
```

Comment:
please use the standard library for this, e.g. `net.ParseIP(canReachHost).To4() != nil` and `net.ParseIP(canReachHost).To16() != nil`